### DISCUSSION TITLE
Install on ARM CPU

### DISCUSSION THREAD

**Discussion by nicopicchio** [jesseduffield/lazygit#2186] [Q&A]
I have a Raspberry Pi 4B running Ubuntu Server 22.04 (headless) to which I connect via SSH through a Windows machine. I tried to install lazygit via apt but the installation fails. I can successfully run lazygit on another Linux machine I have but not on the RPi and I imagine this is because of the RPi's ARM architecture.

Has anyone encountered this issue before and is aware of a workaround please?

**Comment by mark2185** [comment]
> but the installation fails

What error does it throw? 

Have you tried building from source? Namely, just `go run main.go` in the root of the clone will suffice.

EDIT: there was an [issue just like this one](https://github.com/jesseduffield/lazygit/issues/1337) last year, and it does contain a workaround which seems unnecessary, as `mjarkk` pointed out.

**Comment by mowtschan** [comment]
that works for me on Raspberry Pi 4B:
```
uname -a
# Linux raspi 6.6.31+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux

wget https://go.dev/dl/go1.23.0.linux-arm64.tar.gz
sudo rm -rf /usr/local/go
sudo tar -C /usr/local -xzf go1.23.0.linux-arm64.tar.gz
go install github.com/jesseduffield/lazygit@latest
export PATH=$PATH:/usr/local/go/bin:~/go/bin

lazygit -v
# commit=, build date=, build source=unknown, version=unversioned, os=linux, arch=arm64, git version=2.39.2
```

**Reply by electricalgorithm** [reply]
```bash
../go/pkg/mod/github.com/jesseduffield/lazygit@v0.44.1/pkg/config/user_config_validation.go:5:2: package slices is not in GOROOT (/usr/lib/go-1.19/src/slices)
```

Without a Go experience, I'd say it takes longer to install than I expected.

**Reply by mark2185** [reply]
> ```shell
> ../go/pkg/mod/github.com/jesseduffield/lazygit@v0.44.1/pkg/config/user_config_validation.go:5:2: package slices is not in GOROOT (/usr/lib/go-1.19/src/slices)
> ```

You probably need a newer `go` compiler, `go.mod` states `1.22`, and you seem to have `1.19`.

You can also build it on an `x86` host with this:
```shell
GOOS=linux GOARCH=arm64 go build -o lazygit-arm
```
